### PR TITLE
Refine the java client doc

### PR DESCRIPTION
**Pull request by RobertIndie**
Address the following comments:
* https://github.com/apache/pulsar-site/pull/393#discussion_r1098646842
* https://github.com/apache/pulsar-site/pull/393#discussion_r1098649330

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: docs/client-libraries-java-use.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ while (true) {
 }
 ```
 
-If you don't want to block your main thread and rather listen constantly for new messages, consider using a `MessageListener`.
+If you don't want to block your main thread and rather listen constantly for new messages, consider using a `MessageListener`. The `MessageListener` will use a thread pool inside the PulsarClient. You can set the number of threads to use for message listeners in the ClientBuilder.
```
Comment (momo-jun) [inline docs/client-libraries-java-use.md:82]:
```suggestion
If you don't want to block your main thread but constantly listen for new messages, consider using a `MessageListener`. The `MessageListener` uses a thread pool inside the client. You can set the number of threads for message listeners in the ClientBuilder.
```

**Review by momo-jun — APPROVED** [review]
LGTM

**Comment by Anonymitaet** [comment]
@RobertIndie Thanks for your contribution!

Does this change apply to multiple doc versions (which are in the maintenance cycle)? 

If so, please [update them all](https://github.com/apache/pulsar-site/tree/main/versioned_docs).

Thank you! 😊

**Comment by RobertIndie** [comment]
@Anonymitaet, I have updated all of them. PTAL

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**